app/src/main/java/org/akvo/flow/presentation/navigation/FlowNavigation.java

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright (C) 2017 Stichting Akvo (Akvo Foundation)
+ *
+ * This file is part of Akvo Flow.
+ *
+ * Akvo Flow is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Akvo Flow is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Akvo Flow.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.akvo.flow.presentation.navigation;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.design.widget.NavigationView;
+import android.support.v4.app.DialogFragment;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.util.AttributeSet;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.TextView;
+
+import org.akvo.flow.R;
+import org.akvo.flow.app.FlowApp;
+import org.akvo.flow.domain.SurveyGroup;
+import org.akvo.flow.injector.component.ApplicationComponent;
+import org.akvo.flow.injector.component.DaggerViewComponent;
+import org.akvo.flow.injector.component.ViewComponent;
+import org.akvo.flow.ui.Navigator;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+import butterknife.ButterKnife;
+
+public class FlowNavigation extends NavigationView implements FlowNavigationView {
+
+    private TextView currentUserTv;
```

Comment:
Just a suggestion but if these were complete e.g. `currentUserTextView` I would maybe intuitively not have to look at variable list to figure out what is happening in a particular method.

Reply:
OK, I have changed it.